File: src/RepoLite/RepoLite.Tests/ActualGeneratedFIlesTests/GeneratedFiles/AddressDto.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Xml;
using NS.Base;
using NS.Models.Base;

namespace NS.Models
{
	public partial class Address : BaseModel
	{
		public override string EntityName => "Address";
		private Int32 _id;
		private String _anotherid;
		private Int32 _personid;
		private String _line1;
		private String _line2;
		private String _line3;
		private String _line4;
		private String _postcode;
		private String _phonenumber;
		private String _country_code;

		public virtual Int32 Id
		{
			get => _id;
			set => SetValue(ref _id, value);
		}
		public virtual String AnotherId
		{
			get => _anotherid;
			set => SetValue(ref _anotherid, value);
		}
		public virtual Int32 PersonId
		{
			get => _personid;
			set => SetValue(ref _personid, value);
		}
		public virtual String Line1
		{
			get => _line1;
			set => SetValue(ref _line1, value);
		}
		public virtual String Line2
		{
			get => _line2;
			set => SetValue(ref _line2, value);
		}
		public virtual String Line3
		{
			get => _line3;
			set => SetValue(ref _line3, value);
		}
		public virtual String Line4
		{
			get => _line4;
			set => SetValue(ref _line4, value);
		}
		public virtual String PostCode
		{
			get => _postcode;
			set => SetValue(ref _postcode, value);
		}
		public virtual String PhoneNumber
		{
			get => _phonenumber;
			set => SetValue(ref _phonenumber, value);
		}
		public virtual String COUNTRY_CODE
		{
			get => _country_code;
			set => SetValue(ref _country_code, value);
		}
		public override List<ValidationError> Validate()
		{
			var validationErrors = new List<ValidationError>();

			if (string.IsNullOrEmpty(AnotherId))
				validationErrors.Add(new ValidationError(nameof(AnotherId), "Value cannot be null"));
			if (!string.IsNullOrEmpty(AnotherId) && AnotherId.Length > 10)
				validationErrors.Add(new ValidationError(nameof(AnotherId), "Max length is 10"));
			if (string.IsNullOrEmpty(Line1))
				validationErrors.Add(new ValidationError(nameof(Line1), "Value cannot be null"));
			if (!string.IsNullOrEmpty(Line1) && Line1.Length > 100)
				validationErrors.Add(new ValidationError(nameof(Line1), "Max length is 100"));
			if (!string.IsNullOrEmpty(Line2) && Line2.Length > 100)
				validationErrors.Add(new ValidationError(nameof(Line2), "Max length is 100"));
			if (!string.IsNullOrEmpty(Line3) && Line3.Length > 100)
				validationErrors.Add(new ValidationError(nameof(Line3), "Max length is 100"));
			if (!string.IsNullOrEmpty(Line4) && Line4.Length > 100)
				validationErrors.Add(new ValidationError(nameof(Line4), "Max length is 100"));
			if (string.IsNullOrEmpty(PostCode))
				validationErrors.Add(new ValidationError(nameof(PostCode), "Value cannot be null"));
			if (!string.IsNullOrEmpty(PostCode) && PostCode.Length > 15)
				validationErrors.Add(new ValidationError(nameof(PostCode), "Max length is 15"));
			if (!string.IsNullOrEmpty(PhoneNumber) && PhoneNumber.Length > 20)
				validationErrors.Add(new ValidationError(nameof(PhoneNumber), "Max length is 20"));
			if (!string.IsNullOrEmpty(COUNTRY_CODE) && COUNTRY_CODE.Length > 2)
				validationErrors.Add(new ValidationError(nameof(COUNTRY_CODE), "Max length is 2"));

			return validationErrors;
		}
		internal static List<ColumnDefinition> Columns => new List<ColumnDefinition>
		{
			new ColumnDefinition("Id", typeof(System.Int32), "[INT]", SqlDbType.Int, false, true, true),
			new ColumnDefinition("AnotherId", typeof(System.String), "[NVARCHAR](10)", SqlDbType.NVarChar, false, true, false),
			new ColumnDefinition("PersonId", typeof(System.Int32), "[INT]", SqlDbType.Int, false, false, false),
			new ColumnDefinition("Line1", typeof(System.String), "[NVARCHAR](100)", SqlDbType.NVarChar, false, false, false),
			new ColumnDefinition("Line2", typeof(System.String), "[NVARCHAR](100)", SqlDbType.NVarChar, true, false, false),
			new ColumnDefinition("Line3", typeof(System.String), "[NVARCHAR](100)", SqlDbType.NVarChar, true, false, false),
			new ColumnDefinition("Line4", typeof(System.String), "[NVARCHAR](100)", SqlDbType.NVarChar, true, false, false),
			new ColumnDefinition("PostCode", typeof(System.String), "[NVARCHAR](15)", SqlDbType.NVarChar, false, false, false),
			new ColumnDefinition("PhoneNumber", typeof(System.String), "[NVARCHAR](20)", SqlDbType.NVarChar, true, false, false),
			new ColumnDefinition("COUNTRY_CODE", typeof(System.String), "[NVARCHAR](2)", SqlDbType.NVarChar, true, false, false),
		};
	}
}

File: src/RepoLite.VB.Tests/ActualGeneratedFileTests/GeneratedFiles/AddressDto.vb
Imports RepoLite.VB.Tests.MODELNAMESPACE.Base
Imports RepoLite.VB.Tests.REPOSITORYNAMESPACE.Base

Namespace NS.Models
    Partial Public Class Address
        Inherits BaseModel

        Public Overrides ReadOnly Property EntityName As String = "Address"

        Private _id As Int32
        Private _anotherId As String
        Private _personId As Int32
        Private _line1 As String
        Private _line2 As String
        Private _line3 As String
        Private _line4 As String
        Private _postCode As String
        Private _phoneNumber As String
        Private _cOUNTRY_CODE As String

        Public Overridable Property Id As Int32
            Get
                Return _id
            End Get
            Set(value As Int32)
                SetValue(_id, value)
            End Set
        End Property

        Public Overridable Property AnotherId As String
            Get
                Return _anotherId
            End Get
            Set(value As String)
                SetValue(_anotherId, value)
            End Set
        End Property

        Public Overridable Property PersonId As Int32
            Get
                Return _personId
            End Get
            Set(value As Int32)
                SetValue(_personId, value)
            End Set
        End Property

        Public Overridable Property Line1 As String
            Get
                Return _line1
            End Get
            Set(value As String)
                SetValue(_line1, value)
            End Set
        End Property

        Public Overridable Property Line2 As String
            Get
                Return _line2
            End Get
            Set(value As String)
                SetValue(_line2, value)
            End Set
        End Property

        Public Overridable Property Line3 As String
            Get
                Return _line3
            End Get
            Set(value As String)
                SetValue(_line3, value)
            End Set
        End Property

        Public Overridable Property Line4 As String
            Get
                Return _line4
            End Get
            Set(value As String)
                SetValue(_line4, value)
            End Set
        End Property

        Public Overridable Property PostCode As String
            Get
                Return _postCode
            End Get
            Set(value As String)
                SetValue(_postCode, value)
            End Set
        End Property

        Public Overridable Property PhoneNumber As String
            Get
                Return _phoneNumber
            End Get
            Set(value As String)
                SetValue(_phoneNumber, value)
            End Set
        End Property

        Public Overridable Property COUNTRY_CODE As String
            Get
                Return _cOUNTRY_CODE
            End Get
            Set(value As String)
                SetValue(_cOUNTRY_CODE, value)
            End Set
        End Property

        Public Overrides Function Validate() As List(Of ValidationError)
            Dim validationErrors = New List(Of ValidationError)()
            If String.IsNullOrEmpty(AnotherId) Then validationErrors.Add(New ValidationError(NameOf(AnotherId), "Value cannot be null"))
            If Not String.IsNullOrEmpty(AnotherId) AndAlso AnotherId.Length > 10 Then validationErrors.Add(New ValidationError(NameOf(AnotherId), "Max length is 10"))
            If String.IsNullOrEmpty(Line1) Then validationErrors.Add(New ValidationError(NameOf(Line1), "Value cannot be null"))
            If Not String.IsNullOrEmpty(Line1) AndAlso Line1.Length > 100 Then validationErrors.Add(New ValidationError(NameOf(Line1), "Max length is 100"))
            If Not String.IsNullOrEmpty(Line2) AndAlso Line2.Length > 100 Then validationErrors.Add(New ValidationError(NameOf(Line2), "Max length is 100"))
            If Not String.IsNullOrEmpty(Line3) AndAlso Line3.Length > 100 Then validationErrors.Add(New ValidationError(NameOf(Line3), "Max length is 100"))
            If Not String.IsNullOrEmpty(Line4) AndAlso Line4.Length > 100 Then validationErrors.Add(New ValidationError(NameOf(Line4), "Max length is 100"))
            If String.IsNullOrEmpty(PostCode) Then validationErrors.Add(New ValidationError(NameOf(PostCode), "Value cannot be null"))
            If Not String.IsNullOrEmpty(PostCode) AndAlso PostCode.Length > 15 Then validationErrors.Add(New ValidationError(NameOf(PostCode), "Max length is 15"))
            If Not String.IsNullOrEmpty(PhoneNumber) AndAlso PhoneNumber.Length > 20 Then validationErrors.Add(New ValidationError(NameOf(PhoneNumber), "Max length is 20"))
            If Not String.IsNullOrEmpty(COUNTRY_CODE) AndAlso COUNTRY_CODE.Length > 2 Then validationErrors.Add(New ValidationError(NameOf(COUNTRY_CODE), "Max length is 2"))
            Return validationErrors
        End Function

        Shared Friend ReadOnly Property Columns As List(Of ColumnDefinition)
            Get
                Return New List(Of ColumnDefinition) From {
                    New ColumnDefinition("Id", GetType(System.Int32), "[INT]", SqlDbType.Int, False, True, True),
                    New ColumnDefinition("AnotherId", GetType(System.String), "[NVARCHAR](10)", SqlDbType.NVarChar, False, True, False),
                    New ColumnDefinition("PersonId", GetType(System.Int32), "[INT]", SqlDbType.Int, False, False, False),
                    New ColumnDefinition("Line1", GetType(System.String), "[NVARCHAR](100)", SqlDbType.NVarChar, False, False, False),
                    New ColumnDefinition("Line2", GetType(System.String), "[NVARCHAR](100)", SqlDbType.NVarChar, True, False, False),
                    New ColumnDefinition("Line3", GetType(System.String), "[NVARCHAR](100)", SqlDbType.NVarChar, True, False, False),
                    New ColumnDefinition("Line4", GetType(System.String), "[NVARCHAR](100)", SqlDbType.NVarChar, True, False, False),
                    New ColumnDefinition("PostCode", GetType(System.String), "[NVARCHAR](15)", SqlDbType.NVarChar, False, False, False),
                    New ColumnDefinition("PhoneNumber", GetType(System.String), "[NVARCHAR](20)", SqlDbType.NVarChar, True, False, False),
                    New ColumnDefinition("COUNTRY_CODE", GetType(System.String), "[NVARCHAR](2)", SqlDbType.NVarChar, True, False, False)
                    }
            End Get
        End Property

    End Class
End Namespace
